tests/test_schedule/test_alns.py

```diff
@@ -0,0 +1,118 @@
+
+
+from celest.schedule.alns import ALNS
+from unittest import TestCase
+import numpy as np
+
+
+class TestALNS(TestCase):
+
+    def setUp(self):
+
+        self.alns = ALNS([1, 1, 1])
+
+        self.alns.destroy_weights = [1, 2, 3]
+        self.alns.destroy_functions = [0, 0, 0]
+
+        self.alns.repair_weights = [1, 2, 3, 4]
+        self.alns.repair_functions = [0, 0, 0, 0]
+
+    def test_add_cos_func(self):
+
+        self.alns.add_cost_function(True)
+        self.assertTrue(self.alns.cost_function)
+
+    def test_add_destroy_functions(self):
+
+        self.alns.add_destroy_functions([True])
+        self.assertTrue(np.all(self.alns.destroy_functions))
+
+    def test_add_repair_functions(self):
+
+        self.alns.add_repair_functions([True])
+        self.assertTrue(np.all(self.alns.repair_functions))
+
+    def test_add_completeness_function(self):
+
+        self.alns.add_completeness_function(True)
```

Comment:
Consider moving the cost, destroy, repair, and completeness functions to the `setUp` method and actually use them here and in the other methods that use them.

Reply:
Overly complicated.